R/plot_sankey.R

```diff
@@ -1,37 +1,63 @@
 #' Create a sankey plot
 #'
 #' @param data A data frame like [toy_data].
+#' @param with_company Logical. If TRUE, will plot a node with the company name.
+#' If FALSE, will plot without the company name node.
 #'
 #' @return A sankey plot of class [sankeyNetwork].
 #' @export
 #' @importFrom rlang .data
 #'
 #' @examples
 #' plot_sankey(toy_data)
-plot_sankey <- function(data) {
+plot_sankey <- function(data, with_company = TRUE) {
   data_links <- data |>
     mutate(
       source = .data$bank,
       target = .data$pctr_risk_category,
       value = .data$amount,
+      middle_node1 = .data$company_name,
       middle_node2 = .data$tilt_sec
     )
 
-  links <- data_links |>
-    select(
-      "bank",
-      source = "bank",
-      target = "middle_node2",
-      value = "amount",
-      group = "pctr_risk_category"
-    )
+  if(with_company){
+
+    links <- data_links |>
+      select(
+        "bank",
+        source = "bank",
+        target = "middle_node1",
+        value = "amount",
+        group = "pctr_risk_category"
+      )
+
+    links <- data_links |>
+      select(
+        "bank",
+        source = "middle_node1",
+        target = "middle_node2",
+        value = "amount",
+        group = "pctr_risk_category"
+      ) |>
+      bind_rows(links)
+  }else{
+
+    links <- data_links |>
+      select(
+        "bank",
+        source = "bank",
+        target = "middle_node2",
+        value = "amount",
+        group = "pctr_risk_category"
+      )
+  }
 
   links <- data_links |>
     select(
       "bank",
       source = "middle_node2",
       target = "pctr_risk_category",
-      value = "amount_of_disctinct_products",
+      value = "amount",
       group = "pctr_risk_category"
     ) |>
     bind_rows(links)
```

Comment:
Here again. This pattern appears 4 times in this function.

Note that most of this function deals with data preparation. But this is a plotting function so most of it should be about plotting. This suggests it'll be best to extract data preparation elsewhere. Not sure if you would export that so that the user does the preparation, or if you would call the helper from the plotting function. 

```r
prepare_data(data) |> plot_sankey()

# Or
plot_sankey <- function(...) {
  prepare_data(data) |> plot_sankey_impl()
}
plot_sankey(data)
```

You may see 2dii.plot for inspiration about design.